### ISSUE TITLE
Catch 22: SingleModel init checks suffix_to_model_mapping immediately after definition

### ISSUE THREAD

**Issue by simpsus** [crowdcent/numerblox#130]
I am trying to subclass SingleModel with another extension.

Before I call the `SingleModel.__init__ `there is `suffix_to_model_mapping` dict defined, so I cannot add the new extension. If I define it myself, the `SingleModel.__init__` overrides it. If I do not create it, the super init executes` __check_valid_suffix` which fails because my extension is not registered.

_Checkmate._

**Comment by CarloLepelaars** [comment]
Hi Bastian, nice catch! Let's see how we can fix this. Do you already have a good solution in mind for this?

If it is a common format I would also be happy to add it to the `suffix_to_model_mapping`. We are trying to integrate numerblox processors and models more seamlessly with `scikit-learn` so happy to hear ideas.

**Comment by simpsus** [comment]
One Idea I had was to have the __init__ check if `suffix_to_model_mapping` is already present and update it if so

**Comment by CarloLepelaars** [comment]
Good idea, happy to review a PR if you want to implement this and it passes tests. Also was thinking it would be solid to create a (`_add_suffix`) setter method so it's more straightforward to use custom suffixes and model loaders in the future.
